tests/smt2/qf_ufbv/test-attr.smt2

```diff
@@ -0,0 +1,16 @@
+(set-logic QF_UFBV)
+(set-option :produce-models true)
+(set-option :incremental true)
+(declare-const x (_ BitVec 8))
+(declare-const y (_ BitVec 8))
+(define-fun xlow () (_ BitVec 4) ((_ extract 3 0) x))
+(define-fun ylow () (_ BitVec 4) ((_ extract 3 0) y))
+(define-fun xhigh () (_ BitVec 4) ((_ extract 7 4) x))
+(define-fun yhigh () (_ BitVec 4) ((_ extract 7 4) y))
+(declare-fun f ((_ BitVec 8)) (_ BitVec 8))
+(assert (! (distinct (f x) (f y)) :name diff))
+(assert (! (= xlow ylow) :name equal-low))
```

Comment:
Is this iteration supposed to support arbitrary attributes? If so, perhaps we should add more examples to this test, like quantifier patterns, that use terms and not only strings. 
Perhaps we can just use this example from `cvc5`'s regressions as is:
https://github.com/cvc5/cvc5/blob/master/test/regress/regress0/quantifiers/double-pattern.smt2

Reply:
Ahh interesting, I was wondering what `sexprs` in attributes were for. So, this should currently parse but I kept it as a string. So in this case it would get the string literal `(P x)`, as opposed to the term `P` applied to `x`. Do you think I should interpret it as a term in the parser and have a `term_attribute` method that takes a term instead of a keyword and value pair?

Reply:
Hey, I know this has already been approved but I wanted to revisit this discussion. Also interested in feedback from @4tXJ7f. Should s-expr in attributes be processed as terms instead of strings?

Thanks!